torch/_dynamo/variables/builder.py

```diff
@@ -927,7 +932,9 @@ def wrap_tensor(self, value: torch.Tensor):
             tx=self.tx,
             proxy=tensor_proxy,
             example_value=value,
-            guards=self.make_guards(GuardBuilder.TENSOR_MATCH),
+            guards=self.make_guards(
+                functools.partial(GuardBuilder.TENSOR_MATCH, value=value)
```

Comment:
So are you passing value for EVERYBODY, not just ndarray?

Reply:
Yes, this actually saves us an eval, and should always be sound.